### DISCUSSION TITLE
Acceleration rebalance - creating a gameplay niche for small ships, while keeping large ships fun.

### DISCUSSION THREAD

**Discussion by Quantumshark** [endless-sky/endless-sky#8957] [Balance, Mechanics, and Interactions]
### Concept
So, as most people agree, currently most small ships don't really offer anything late-game that larger ships can't do better. At the same time, it's very much true that slow ships aren't fun, and ships that are too fast to control aren't really fun either. So, our goal should be to find a way to give small ships a viable gameplay niche, without either making small ships too fast to control, or making big ships too slow to be fun. To reiterate this point, _Big ships should still be fun._ Making big ships unpleasant to play in order to force people into smaller ones is not, actually, a good approach. Instead, the bulk of the changes should be giving smaller ships advantages that they don't currently have. I have made a small step in this direction in #8842, which focuses on tweaking top speeds into a more favourable environment and giving smaller ships an edge, but I don't think #8842 alone really makes smaller ships _viable._

After extensive meddling with the three mobility statistics (top speed, acceleration, and turn rate), I have personally found that acceleration can be varied a _lot_ more than the other two in either direction without becoming unenjoyable. Turning is limited to the most part to the range ~50 - ~200, and even those extremes are questionable. Top speed becomes awful below about 200-300, and difficult to control past about 1500-2000. Acceleration, on the other hand, can happily be anywhere from 50 to 3000 and, provided the other two stats are in sensible relative positions, is still enjoyable. Even better, it really contributes to the _feel_ of a ship. A heavy warship can have 500 top speed, 80 acceleration and 100 turning, and the result is a ship that's fun to play, but _feels_ heavy. A lighter ship can have the same 500 top speed and 100 turning with 500 acceleration, and the result is a very different experience. Even without an actual advantage in top speed or turning, a ship with such a big acceleration advantage can easily keep a distance from a larger ship with lower acceleration.

So, my goal is this: tweak acceleration:top speed ratios across the board, giving ratios of ~2:1 for the smallest ships (fighters and similar), and ~1:5 for the larger end (large heavy warships like the Carrier and Dreadnought; superheavies could be even lower).

This would also solve the issue of the large engines marauders (Leviathan and Falcon), and potential future alien ships with similar characteristics. Currently, small fast ships don't do anything that such ships can't, but with this change those ships might match smaller ones for top speed and turning, but they'd still be massively different in acceleration. Thus, the niche of the big, fast ship can safely exist without obsoleting the small, fast ship.

Also, this would make it much easier to avoid large pirate warships in all three of the starting ships, which is something the early game would benefit from. 

### Proposed implementation
This is where this gets slightly complicated. Currently, turning and acceleration are both tied to mass, so they can't be independently varied at the level of the ship. But that's OK, because we can change the stats of engines.
So, here's the full plan.
1. Make the size scaling on steering outfits _steeper_. Probably by a factor of about three, with the smallest engines being invariant. _Do not make the same adjustment to thrusters._

2. Adjust the hull mass of every ship in the game to keep their turn rates unchanged. This can probably be handled systematically, just by looking at the stock engines, though there will doubtless be edge cases.

3. Apply a flat multipler to thruster thrust and ship drag across the board, probably a factor of 4. The previous steps will leave smaller ships unchanged - their mass can't be dropped by much without going negative, so increasing masses for large ships is the approach which is possible. However, this leaves small ships with the current acceleration : top speed ratios, rather than the much higher ones that I'm aiming for to give them quick, responsive handling. This step simply shifts acceleration : top speed ratios across the board, and can be calibrated based on what the results of step 3 look like in practice.

4. (optional) Adjust heat dissipations so that max heat is unchanged. This I could go either way on, but it might limit the number of outfits expansions that can be added to ships like the Bactrian and Rano'erek. Whether that's a good thing or a bad thing is up to you.

5. (optional) Rework the handling of heavy freighters. Currently, heavy freighters have some of the highest acceleration : max speed ratios in the game, thanks to overinflated drag stats intended to hamper their mobility. I'd rather see the reverse - freighters that _can_ get up to decent speeds, but have minimal ability to stop and change course. Almost like a speeding freight train, in fact. Large freighters already have low engine space values for their masses, so would end up with fairly low acceleration stats anyways, so all this would require is adjusting their drag.

**Comment by lumbar527** [comment]
This sounds really cool and we do need small ships more appealing!

**Comment by Saturnalia9** [comment]
Not sure about the smaller ships simply keeping their distance. Smaller ships should be able to outrun bigger ships. With each tier (heavy, medium, light, etc.) becoming faster than the previous. For instance, an interceptor enters a system and has to accelerate up to max speed while the heavy is already at max. No advantage there. 

I assume that was just an example, though.

**Reply by Quantumshark** [reply]
You can do a lot more with extremely high acceleration than just keep your distance - I included that point to clarify that this would also be useful for small noncombatants, which tend not to be very fast and aren't going to benefit from the kinds of maneuvering that could give small warships a combat advantage. Currently, there's no way a Star Barge or even a Shuttle with the stock ion engines can outrun a Falcon built for speed, but with this change, while they may still not be able to infinitely outrun that Falcon in a straight line, they could easily outmaneuver it in-system, keep a distance, and safely land.

Even if you scrapped the concept of big fast ships, there's still the question of "How do you give small slow ships, things like the Modified Argosy or even the Berserker, a niche if the only advantage small ships get is speed?" While I agree that the overall trend should be decreasing speed with increasing size, there is always going to be some overlap in top speed across size categories, because any ship can be relatively fast or relatively slow for its size - my overall goal here is to create something which gives small ships an advantage, is inextricably tied to size, but still leaves us with all our current variety between ships of similar sizes.

**Reply by Saturnalia9** [reply]
Yeah, some larger ships will be faster, but bringing them a bit more in line (so that Falcons don't outrun all but the fastest mid-sized ships) should be addressed a bit as well.

**Reply by Quantumshark** [reply]
I agree. #8842 makes some initial steps on that, I think that combined with the acceleration changes here would at least be enough of a change together that they'd be worth testing. There's always room to up the Falcon's drag a little if it's still needed.

**Reply by Saturnalia9** [reply]
Really the biggest reason why I want to keep all ships in one category a similar speed is so that it's not just "use [specific ship] as your flag because it's the last [unbalanced] fast heavy warship".

**Reply by Saturnalia9** [reply]
Something else worth noting is that smaller ships should have enough speed that they're mobile enough to "outrun" missiles, but still be slow enough to not outrun guns.

Likewise, the smallest ships should still be able to drastically outrun the heaviest ships. No idea how that would be addressed, but... that might be a good start, to at least stsrt getting close to these scenarios.

**Reply by Ferociousfeind** [reply]
if missiles are supposed to be outran, they should be relabeled- a missile's main bonus is being guided but not having a squishy fragile human in the projectile doing the guiding. They are nightmares because they can pull 45G maneuvers in air to maintain an interception trajectory on their target. If you, the squishy human, can just outrun missiles, what's really the point of them?

**Reply by Saturnalia9** [reply]
> if missiles are supposed to be outran, they should be relabeled- a missile's main bonus is being guided but not having a squishy fragile human in the projectile doing the guiding. They are nightmares because they can pull 45G maneuvers in air to maintain an interception trajectory on their target. If you, the squishy human, can just outrun missiles, what's really the point of them?

Our missiles don't have that capability of pulling 45G maneuvers. Yet somehow our fighters and some interceptors can. So I'm assuming that wasn't a game flaw, but something done on purpose.

**Reply by Ferociousfeind** [reply]
huh...? plenty of missiles (in ES) have really high acceleration, and plus drag can radically increase their effective visual acceleration. Slap homing, uhh, 3, on a missile, and it'll turn around without accelerating constantly (to give it a tighter turning circle), which will, when it's accelerating directly against its current velocity vector, will, plus drag, give it a HUGE momentary acceleration, which I'd say is akin to 45G maneuvers you'll see Tomahawks, or whichever missiles intercept airborne fighters in real life, pull. And our fighters are sluggish compared to what a "fighter" even in an ES context should be able to do- they're just so big and chunky (and also fragile) and slow. Bottom line is, if they're outrunning missiles, something is very wrong.

But, you're right from a gameplay standpoint, missiles that are unavoidable are realistic but not that fun. There's gotta be SOME way to hollywood michael bay your way out of getting blown out of the sky by a pair of missiles, otherwise they're nothing more than hitscan, or an AoE attack.

I think that straight-line outrunning them is the WORST solution to allow. There is _zero_ skill involved in that- if you just keep moving, missiles will never be a problem. I'd rather they have sizable turning circles, so you can foil missiles by getting into those circles and waiting for them to expire, or lose target and fly off.

**Reply by Saturnalia9** [reply]
What I meant is that missiles don't have an extremely high turn stat. If they did, that would be closer to the 45G angular acceleration. I wouldn't mind having extremely fast weak missiles with low turning, to use against fighters. Then you could more effectively have fighters dodge missiles (by even a small amount of turning) or getting hit (by not trying to dodge at all). 

Most missiles (aside from torpedos) have decent speed and decent turn. But not more of one or the other.

**Reply by Zitchas** [reply]
I figure that everything should have a counter:

Torpedoes are generally not very agile: Can be outrun or out maneuvered.
Heavy rockets don't track, so they can be dodged.
Javelins don't hit too hard, so they can be tanked fairly easily.

Missiles are less clear, though. We have one missile that is clearly described as being a highly agile weapon, which sounds like something to be used for highly agile targets. That one, at least, should be quite agile and extremely hard to dodge. Jamming should probably be the way to deal with that.

> Although not as powerful or cheap as the popular Meteor Missiles, Sidewinders have several advantages, including their superior maneuverability, better tracking systems, and protective casings that make them much harder for anti-missile systems to destroy.


The other human missile, the meteor, should probably just match its description:
> The Meteor Missile is the cheapest and simplest homing weapon available in human space, with only rudimentary infrared tracking systems that can have trouble homing in on targets with low heat levels. True to its name, the missile reaches a very high speed to leave its target's anti-missile defenses with little time to react. The trade-off is maneuverability, so Meteor Missiles are practically useless against agile targets farther than point-blank range.

That sounds like something that should be easy for small ships to dodge; but almost impossible to outrun. I will admit, though, I don't like the second part where it talks about being a point-blank weapon.

I do wish we had flares. They were a lot of fun to use in EV as a cheap and effective way for small ships to deal with individual IR weaponry. (but fairly useless against large numbers)

Either way, the important part is to ensure that there is a counter to them. Guns are the primary weapons that should be doing the consistent damage, secondaries as a whole should have counters of some kind and generally be higher risk / higher reward options. (risk, in this case, being the risk that the target avoids the damage and/or hits allies.)

**Reply by Saturnalia9** [reply]
> Guns are the primary weapons that should be doing the consistent damage, secondaries as a whole should have counters of some kind and generally be higher risk / higher reward options. (risk, in this case, being the risk that the target avoids the damage and/or hits allies.)

If missiles had lower numbers and hit harder (and then got balanced off those two stats), that would be my idea of a decent risk-reward system. Like holding a max of 10 missiles with decent initial turning, high max speed, and low acceleration to counter slower heavier ships. And then high acceleration, relatively mediocre max speed, and low turning to use against smaller ships.

**Reply by Zitchas** [reply]
You might be interested to check out the "Revamp" PR from a while back: https://github.com/endless-sky/endless-sky/pull/5772 . It encompassed a lot of these changes and more as part of a holistic revamping of the combat system. Overall it was fairly well received, but not in its entirety. In any case, you may be interested at looking at the things that were tested out there and the discussion that was had about it.

**Reply by Saturnalia9** [reply]
> You might be interested to check out the "Revamp" PR from a while back: https://github.com/endless-sky/endless-sky/pull/5772 . It encompassed a lot of these changes and more as part of a holistic revamping of the combat system. Overall it was fairly well received, but not in its entirety. In any case, you may be interested at looking at the things that were tested out there and the discussion that was had about it.

Already know about iit, had been using it for some time (like a year or so). It's fairly outdated, which is why I stopped using it.

> But, you're right from a gameplay standpoint, missiles that are unavoidable are realistic but not that fun. There's gotta be SOME way to hollywood michael bay your way out of getting blown out of the sky by a pair of missiles, otherwise they're nothing more than hitscan, or an AoE attack.

Having missiles that go fast, have low turning, and have homing 4 (so they don't need as much turning since it's intercepting) was my intention. Provided you can see it coming (and watch for it), you can easily dodge it. But if not, it punishes you.

**Comment by TheGiraffe3** [comment]
Should this discussion be closed, seeing as how #9616 was merged?

**Reply by Quantumshark** [reply]
This is a slightly different idea, but I think you're probably right that we don't need both this and #9616
